Code/Tools/AssetProcessor/native/ui/BuilderInfoMetricsModel.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AZCore/std/containers/vector.h>
+#include <AZCore/std/containers/unordered_map.h>
+#include <AzCore/std/string/string.h>
+#include <QAbstractItemModel>
+#include <QSortFilterProxyModel>
+#include <AzToolsFramework/AssetDatabase/AssetDatabaseConnection.h>
+
+namespace AssetBuilderSDK
+{
+    struct AssetBuilderDesc;
+} // namespace AssetBuilderSDK
+
+namespace AssetProcessor
+{
+    class BuilderDataItem;
+    class BuilderData;
+    class JobEntry;
+
+    class BuilderInfoMetricsModel
+        : public QAbstractItemModel
+    {
+    public:
+        enum class Column
+        {
+            Name,
+            JobCount,
+            TotalDuration,
+            AverageDuration,
+            Max
+        };
+
+        enum class Role
+        {
+            SortRole = Qt::UserRole
+        };
+
+        BuilderInfoMetricsModel(BuilderData* builderData, QObject* parent = nullptr);
+        void Reset();
+        void OnBuilderSelectionChanged(const AssetBuilderSDK::AssetBuilderDesc& builder);
+
+        // QAbstractItemModel
+        QModelIndex index(int row, int column, const QModelIndex& parent = QModelIndex()) const override;
+        int rowCount(const QModelIndex& parent = QModelIndex()) const override;
+        int columnCount(const QModelIndex& parent = QModelIndex()) const override;
+        QVariant data(const QModelIndex& index, int role = Qt::DisplayRole) const override;
+        QVariant headerData(int section, Qt::Orientation orientation, int role = Qt::DisplayRole) const override;
+        QModelIndex parent(const QModelIndex& index) const override;
+
+    public Q_SLOTS:
+        void OnDurationChanged(BuilderDataItem* item);
+
+    private:
+        BuilderData* m_data;
```

Comment:
Can you use a smart pointer of some type here, maybe a shared pointer? If not, can you initialize to nullptr here?